### DISCUSSION TITLE
Not getting ClientID

### DISCUSSION THREAD

**Discussion by CosmicTorpedo** [cmu-sei/GHOSTS#274] [Q&A]
I've been playing around with GHOSTS and works great until I start playing around with it. None of the may GHOSTS clients can communicate to the GHOSTS API. The possible error I get in the logs is: 

"Ghosts.Client.Comms.CheckId.get_Id|Skipping check for ID from server, too many requests in a short amount of time..."

I've done clean installs on new machines and old and even blew the away the docker containers to and redeployed and still have the same issue. Any suggestions or ideas on why the clients won't communicate to the API server?

<img width="862" alt="pic1" src="https://github.com/cmu-sei/GHOSTS/assets/160567102/93f98d7a-c43a-416e-9a97-df901e40bbed">

**Comment by sei-dupdyke** [comment]
Thanks for checking out the project.

Can your clients get to the API? A simple check is to try and hit the API url in the browser. If you can, then next check the client logs at ./logs/application.log to see if there are errors.

**Comment by CosmicTorpedo** [comment]
The clients can reach the API via the  <IP>:5000/index.html ok. I researched that the id.josn should be created in the instance folder but no id.json is created there and I'm thinking that all the communication is dependent on that to communicate with the API. 

I do not get and errors, per se, but these are the only messages that seem to be the culprit in the app.log:: (granted I did lower the CycleSleep to be like 3 seconds, regardless even left at default it still said the same message)
<img width="884" alt="pic1" src="https://github.com/cmu-sei/GHOSTS/assets/160567102/0bf3b194-26dc-4843-bdb2-a9cd72e35b8a">

**Reply by sei-dupdyke** [reply]
If there is no id.json file, then the client has not checked in with the API. I have not seen the path format error — it should be in the form http://some.server.com:some_port/api and you should be able to hit that url in your client machine browser to verify the connection can be made.

**Comment by CosmicTorpedo** [comment]
I originally left out the 'http://'. I put the "http://' with the server IP and it looks like it was able to communicate with the server.
